config/initializers/extensions.rb

```diff
@@ -0,0 +1,8 @@
+class Array
+  # Use one_and_only instead of first when dealing with arrays that
+  # should only ever have one element. 
+  def one_and_only
+    raise "Not an array with a single element" unless size == 1
+    first
+  end
```

Comment:
I'm not convinced by this.  I think we should be making the frontend more defensive against unexpected data.  This will lead to 500 errors.  There are cases (particularly while migrating) where the "singular" links items could be unset, or set to empty array.

Also I think adding methods to core classes should be avoided if at all possible.